webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -375,6 +404,204 @@ const AdvanceTextEditor = ({
         }
     }, [message, input]);
 
+    const handleKeyDown = (e: React.KeyboardEvent<TextboxElement>) => {
```

Comment:
0/5 on changing this now since it could be done later on as well, but I wonder if we could pull all this logic out into some sort of `useEditorHotkeys` hook that encapsulates a bunch of logic and just returns the key handler. That's purely hypothetical though since I've never tried that sort of thing

Reply:
My only fear with this (I actually wanted to move this all out of the component directly) is that uses several "component variables" (state, props, etc...). So we may end up with a hook with many arguments. We can take a look in the future how to simplify this further once we have gotten rid of the `advanced_create_comment` and `advanced_create_post`.